src/scripting/lua_widget_attributes.cpp

```diff
@@ -96,15 +97,28 @@ static void push_child_by_index(lua_State *L, gui2::widget& w, int i)
 	} else if(gui2::menu_button* menu_button = dynamic_cast<gui2::menu_button*>(&w)) {
 		int n = menu_button->get_item_count();
 		if(i > n) {
-			std::string s = "index out of range (" + std::to_string(i) + " > " + std::to_string(n) + ")";
-			throw std::invalid_argument(s);
+			config c;
+			for(; n < i; ++n) {
+				menu_button->add_row(c);
+			}
 		}
 		luaW_pushmenuitem(L, *menu_button->get_row(i - 1));
+	} else if(gui2::multimenu_button* multimenu_button = dynamic_cast<gui2::multimenu_button*>(&w)) {
+		int n = multimenu_button->get_item_count();
+		if(i > n) {
+			config c;
+			for(; n < i; ++n) {
+				multimenu_button->add_row(c);
+			}
+		}
+		luaW_pushmenuitem(L, *multimenu_button->get_row(i - 1));
```

Comment:
Since the menu_button and multimenu_button cases are identical to options_button, and they are subclasses of options_button, you can just delete them here – they'll then be handled by the options_button case below.